apps/frontend/src/pages/root/components/user-wheel/user-wheel.tsx

```diff
@@ -43,22 +48,55 @@ const UserWheel: React.FC = () => {
 		setIsEditingModalOpen(false);
 	}, []);
 
+	const handleModeToggle = useCallback(() => {
+		setEditMode((previousState) => {
+			return previousState === "manual" ? "retake_quiz" : "manual";
+		});
+	}, []);
+
 	useEffect(() => {
 		void dispatch(quizActions.getScores());
 	}, [dispatch]);
 
+	function getModal(mode: WheelEditMode): React.ReactNode {
+		switch (mode) {
+			case "manual": {
+				return (
+					<ScoresEditModal data={scores} onSaveChanges={handleFinishEditing} />
+				);
+			}
+
+			case "retake_quiz": {
+				return <RetakeQuizModal />;
+			}
+
+			default: {
+				return;
+			}
+		}
+	}
+
 	return (
 		<div className={styles["container"]}>
 			<div className={styles["header"]}>
 				<h4 className={styles["header-text"]}>{headerText}</h4>
+				{isEditingModalOpen && (
+					<Switch
+						currentMode={editMode}
+						leftButtonProperties={{ label: "Edit manually", mode: "" }}
```

Comment:
I guess that mode here should not be an empty

Reply:
Didn't quite understand what you mean

Reply:
Is it ok that `mode: ""`? Or maybe should it be `mode: "edit_manually"`?